### ISSUE TITLE
ui-router params

### ISSUE THREAD

**Issue by bathos** [olov/ng-annotate#207]
Although ngAnno is aware of $stateProvider's resolves, it doesn't seem to pick up params. The values in the params object, like resolve, can be functions that have injections. Since in other regards ngAnno "knows about" ui-router, it seems like this might just be an oversight, though maybe it was a deliberate choice since the feature is not used as much; figured I'd report it just in case though.

http://angular-ui.github.io/ui-router/site/#/api/ui.router.state.$stateProvider#params

```
$stateProvider.state({
  name: 'foo',
  url: '/foo?bar',
  params: {
    bar: $baz => $baz.getFallbackBar()
  },
  /* etc */
});
```

It can also take the form `params: { bar: { value: INJECTABLE } }`.

**Comment by olov** [comment]
Just to clarify, is it guaranteed that all entries in the params object, if they are functions, also are injectable?

**Comment by bathos** [comment]
As far as I'm aware, yes. The values can be strings or functions (injectable) that return strings.

I am unsure whether these are currently supported or not, but it might be worth mentioning that there are still other injectable properties in that state config: templateProvider, controllerProvider, onEnter, and onExit.

**Comment by olov** [comment]
released in 1.1.0
